api/database/seeders/DatabaseSeeder.php

```diff
@@ -87,16 +87,60 @@ public function run()
         $pool = Pool::whereNotNull('published_at')->inRandomOrder()->first();
         $this->seedPoolCandidate($applicant, $pool);
         $this->seedAwardExperienceForPool($applicant, $digitalTalentPool);
-        $applicantUserSkills = $applicant->userSkills;
+        $applicantUserSkills = $applicant->userSkills->load(['skill']);
+        $applicantUserSkillTechnicalIndex = 0;
+        $applicantUserSkillBehaviouralIndex = 0;
         foreach ($applicantUserSkills as $applicantUserSkill) {
-            $isSkillLevelSet = $faker->boolean(75);
-            $isWhenSkillUsedSet = $faker->boolean(75);
-            if ($isSkillLevelSet) {
+            if ($faker->boolean(75)) {
                 $applicantUserSkill->skill_level = $faker->randomElement(ApiEnums::skillLevels());
             }
-            if ($isWhenSkillUsedSet) {
+            if ($faker->boolean(75)) {
                 $applicantUserSkill->when_skill_used = $faker->randomElement(ApiEnums::whenSkillUsed());
             }
+            // consistent seeding of applicant's skill rankings that matches what is expected
+            // four types independent of each other, starting from 1, with no duplicates or gaps
+            if ($applicantUserSkill->skill->category === 'TECHNICAL') {
```

Comment:
I find this use of switch/case a bit confusing and unexpected. Could we do all this imperitively outside the foreach loop?
Something more like...
```
foreach {
  ...
}
// Add skills to showcase lists
$applicantUserTechnicalSkills = < user skills where category is technical >
$applicantUserTechnicalSkills[0]->top_skills_rank = 1;
$applicantUserTechnicalSkills[0]->save()
$applicantUserTechnicalSkills[1]->top_skills_rank = 2;
$applicantUserTechnicalSkills[1]->save();
$applicantUserTechnicalSkills[2]->top_skills_rank = 3;
$applicantUserTechnicalSkills[2]->improve_skills_rank = 1;
$applicantUserTechnicalSkills[2]->save();
```

Reply:
Just ignore me if I'm out of the loop and misunderstanding this but could't we just use faker for this?

```php
foreach($skills as $skill) {
  $skill->top_skills_rank = $faker->unique()->numberBetween(1,5); // 5 or length of array 🤔 
  $skill->save():
}
```

Reply:
Okay, I just implemented the above in 
https://github.com/GCTC-NTGC/gc-digital-talent/pull/7678/commits/f1c8681b3d66d9bbd7e4a77eaea5f3c74a6f7cc0